File: 2018/python/day1.py
values = list(map(int, open("../input/day1.input").readlines()))
print "answer a:", sum(values)
f = 0;
freqs = {};
i = 0;
while True:
    f += values[i % len(values)]
    if f in freqs:
        break
    freqs[f] = 0
    i += 1
print "answer b:", f

File: 2018/csharp/day1.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

namespace aoc2018
{
    class Day1
    {
        public static void Solve()
        {
            var values = File.ReadAllLines(Path.Combine("..", "input", "day1.input")).Select(line => int.Parse(line));
            Console.WriteLine("Day 1, answer a: {0}", values.Sum());

            var f = 0;
            var freqs = new Dictionary<int, int>();

            while (true)
            {
                foreach (var value in values)
                {
                    f += value;

                    if (!freqs.ContainsKey(f))
                    {
                        freqs[f] = 0;
                    }
                    else
                    {
                        Console.WriteLine("Day 1, answer b: {0}", f);
                        return;
                    }
                }
            }
        }
    }
}
